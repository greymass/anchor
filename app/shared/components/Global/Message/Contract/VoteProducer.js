// @flow
import React, { Component } from 'react';
import { translate } from 'react-i18next';

import GlobalMessageContract from './Base';

export class GlobalMessageContractVoteProducer extends Component<Props> {
  render() {
    const { data, t } = this.props;
    return (
      <GlobalMessageContract>
        <p>{t('system_voteproducer_1', data)}</p>
        <p>{t('system_voteproducer_2', data)}</p>
        <p>{t('system_voteproducer_3', data)}</p>
        <p>{t('system_voteproducer_4', data)}</p>
        <p>{t('system_voteproducer_5', data)}</p>
      </GlobalMessageContract>
    );
  }
}

export default translate('contracts')(GlobalMessageContractVoteProducer);
