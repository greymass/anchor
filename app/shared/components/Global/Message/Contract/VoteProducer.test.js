import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalMessageContractVoteProducer } from './VoteProducer';

const mockProps = {
  data: {},
  t: jest.fn,
};

describe('GlobalMessageVoteProducer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalMessageContractVoteProducer {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
