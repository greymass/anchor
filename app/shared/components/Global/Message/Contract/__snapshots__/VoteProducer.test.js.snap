// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalMessageVoteProducer renders correctly 1`] = `
<Translate(GlobalMessageContractBase)>
  <p>
    <Component />
  </p>
  <p>
    <Component />
  </p>
  <p>
    <Component />
  </p>
  <p>
    <Component />
  </p>
  <p>
    <Component />
  </p>
</Translate(GlobalMessageContractBase)>
`;
